src/ir/datalog/program.h

```diff
@@ -111,6 +116,11 @@ class Argument {
     return absl::StrCat(argument_name_, " : ", argument_type_.DebugPrint());
   }
 
+  bool operator==(const Argument& otherArgument) const {
+    return this->argument_name_ == otherArgument.argument_name() &&
+    this->argument_type_ == otherArgument.argument_type();
```

Comment:
Use fields directly: 
```suggestion
    return this->argument_name_ == otherArgument.argument_name_ &&
    this->argument_type_ == otherArgument.argument_type_;
```